### PR TITLE
Initial Round Voting Implementation

### PR DESCRIPTION
**Pull request by CodeIsTheKey**
Round Voting initial implementation.  Miner voting tested and working.  Node voting implementation in progress (do not enable).

### CODE REVIEW

File: src/update/update.h [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+         (
+            (bit < 0 || bit > 28)                                         ||
+            (roundSize < 1)                                               ||
+            (startHeight < 0)                                             ||
+            (votingPeriod < 1)                                            ||
+            (votingMaxRounds < 1)                                         ||
+            (graceRounds < 0)                                             ||
+            (heightActivated < -1)                                        ||
+            (startHeight % roundSize != 0)                                ||
+            (votingPeriod > votingMaxRounds)
```
Comment (nandofw) [inline src/update/update.h:110]: is this line correct?
note: it fail on launch as votingPeriod is 4031 and votingMaxRounds is 3

File: src/llmq/quorums_commitment.cpp [hunk 2]
```diff
@@ -86,8 +87,14 @@ namespace llmq {
 
         // sigs are only checked when the block is processed
         if (checkSigs && IsBlsSigCheckEnabled(pQuorumBaseBlockIndex->GetBlockTime())) {
-            uint256 commitmentHash = CLLMQUtils::BuildCommitmentHash(llmq_params.type, quorumHash, validMembers,
-                                                                     quorumPublicKey, quorumVvecHash);
+
+            uint256 commitmentHash;
+            if (UpdateManager::Instance().IsActive(EUpdate::ROUND_VOTING, pQuorumBaseBlockIndex)) {
+                commitmentHash = CLLMQUtils::BuildCommitmentHash(llmqType, quorumHash, validMembers, quorumUpdateVotes, quorumPublicKey, quorumVvecHash);
```
Comment (nandofw) [inline src/llmq/quorums_commitment.cpp]:
```suggestion
                commitmentHash = CLLMQUtils::BuildCommitmentHash(llmq_params.type, quorumHash, validMembers, quorumUpdateVotes, quorumPublicKey, quorumVvecHash);
```

Comment (nandofw) [inline src/llmq/quorums_commitment.cpp]:
```suggestion
                commitmentHash = CLLMQUtils::BuildCommitmentHash(llmq_params.type, quorumHash, validMembers, quorumUpdateVotes, quorumPublicKey, quorumVvecHash);
```

File: src/update/update.cpp [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+   minerUpdateVoting(&minerRoundVoting),
+   nodeUpdateVoting(&nodeRoundVoting)
+{
+}
+
+UpdateManager::~UpdateManager() {};
+
+bool UpdateManager::Add(Update update)
+{
+   // TODO: Check for existence first
```
Comment (nandofw) [inline src/update/update.cpp]:
```suggestion
  //Check for existence first and erase if exist
   auto it = updates.find(update.UpdateId());
   if (it != updates.end())
      updates.erase(it);
```
during start up 4 chains are initialized and the last chain initialized is the active chain, we only need update settings for the active chain

**Review by nandofw — CHANGES_REQUESTED** [review]

File: src/update/update.h (hunk 1 above)
Comment (CodeIsTheKey) [reply src/update/update.h:110]: Good catch - the mainnet votingMaxRounds should be > 4031 (I'll update to 3*4031 to safely cover the range)

File: src/llmq/quorums_commitment.cpp (hunk 2 above)
Comment (CodeIsTheKey) [reply src/llmq/quorums_commitment.cpp]: Will update - logically should be the same, but better to keep consistent.

File: src/update/update.cpp (hunk 3 above)
Comment (CodeIsTheKey) [reply src/update/update.cpp]: Will fix - thanks.

**Comment by CodeIsTheKey** [comment]
Updated from PR comments.  Removed raptoreum-tx from build.yaml, minor typo fix.

**Review by nandofw — APPROVED** [review]

**Comment by CodeIsTheKey** [comment]
Fixed assetindex problem when loading block index.

**Pull request merged**